shell/platform/darwin/ios/rendering_api_selection.mm

```diff
@@ -30,17 +30,20 @@ bool ShouldUseMetalRenderer() {
 #endif  // FLUTTER_SHELL_ENABLE_METAL
 
 IOSRenderingAPI GetRenderingAPIForProcess() {
-#if TARGET_IPHONE_SIMULATOR
-  return IOSRenderingAPI::kSoftware;
-#endif  // TARGET_IPHONE_SIMULATOR
-
 #if FLUTTER_SHELL_ENABLE_METAL
   static bool should_use_metal = ShouldUseMetalRenderer();
   if (should_use_metal) {
     return IOSRenderingAPI::kMetal;
   }
 #endif  // FLUTTER_SHELL_ENABLE_METAL
+
+  // OpenGL will be emulated using software rendering by Apple on the simulator, so we use the
+  // Skia software rendering since it performs a little better than the emulated OpenGL.
+#if TARGET_IPHONE_SIMULATOR
+  return IOSRenderingAPI::kSoftware;
```

Comment:
We could have something above this if `--enable-software-rendering` has been specified to force the emulator to run using software rendering. I think we should do this for parity with Android, and then for now we could make the bots run using software rendering until we're ready to update goldens.

Reply:
An easy way to do this would be to add a parameter to `GetRenderingAPIForProcess` about whether to force software rendering, and pass `settings.enable_software_rendering` as the parameter.

If that's set to true, we just return `IOSRenderingAPI::kSoftware`.